src/pages/index.js

```diff
@@ -0,0 +1,284 @@
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+
+  cardNameEl.textContent = data.name;
+  cardImage.src = data.link;
+  cardImage.alt = data.altText;
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  cardLikeBtn.addEventListener("click", (evt) =>
+    handleLike(evt, data._id, data.isLiked)
+  );
+
+  cardDeleteBtn.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  cardImage.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImage.src = data.link;
+    previewModalImage.alt = data.name;
+    previewModalCaption.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  requestAnimationFrame(() => {
+    document.addEventListener("keydown", closeModalEsc);
+    modal.addEventListener("mousedown", handleOveraly);
+  });
+}
+
+function closeModal(modal) {
+  if (modal) {
+    modal.classList.remove("modal_opened");
+    document.removeEventListener("keydown", closeModalEsc);
+    modal.removeEventListener("mousedown", handleOveraly);
+  }
+}
+
+function handleOveraly(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function closeModalEsc(evt) {
+  if (evt.key === "Escape") {
+    const modal = document.querySelector(".modal_opened");
+    closeModal(modal);
+  }
+}
+
+function handleLike(event, id, isLiked) {
+  api
+    .changeLike(id, isLiked)
+    .then(() => {
+      event.target.classList.toggle("card__like-btn_liked");
+    })
+    .catch(console.error);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  buttonText(submitBtn, true, "Delete", "Deleting...");
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      buttonText(submitBtn, false, "Delete", "Deleting");
+    });
+}
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  buttonText(submitBtn, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+      disableButton(cardSubmitBtn, config);
+    })
+    .catch(console.error)
+    .finally(() => {
+      buttonText(submitBtn, false, "Saving...", "Save");
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  buttonText(submitBtn, true, "Saving...", "Save");
```

Comment:
`"Saving...", "Save"` has wrong **order**. It should be `"Save", "Saving..."`


```js
      buttonText(submitBtn, true, "Save", "Saving...");

```

Please, don't pass the texts into `buttonText` if they coinside with the **default** texts inside the function


```js

      buttonText(submitBtn, true);

```